boot/assets.go

```diff
@@ -322,100 +338,129 @@ func (o *TrustedAssetsInstallObserver) ChosenEncryptionKey(key secboot.Encryptio
 }
 
 // TrustedAssetsUpdateObserverForModel returns a new trusted assets observer for
-// tracking changes to the measured boot assets during gadget updates, provided
-// the device model supports secure boot. Otherwise, nil and ErrObserverNotApplicable is
-// returned.
+// tracking changes to the trusted boot assets and preserving managed assets,
+// provided such support is indicated by the device model. Otherwise, nil and
+// ErrObserverNotApplicable is returned.
```

Comment:
```
// TrustedAssetsUpdateObserverForModel returns a new trusted assets observer for
// tracking changes to the trusted boot assets and preserving managed assets,
// provided the device model indicates this might be needed. 
// Otherwise, nil and ErrObserverNotApplicable is returned.
```